napari/layers/utils/link_layers.py

```diff
@@ -0,0 +1,145 @@
+        include private attributes
+
+    Returns
+    -------
+    names : set of str
+        A set of attribute names that may be linked between ``layers``.
+    """
+    from inspect import ismethod
+
+    try:
+        first_layer = next(iter(layers))
+    except StopIteration:
+        raise ValueError("``layers`` iterable must have at least one layer")
+    common_events = set.intersection(*(set(lay.events) for lay in layers))
+    common_attrs = set.intersection(*(set(dir(lay)) for lay in layers))
+    if not with_private:
+        common_attrs = {x for x in common_attrs if not x.startswith("_")}
+    common = common_events & common_attrs - prohibited
+
+    # lastly, discard any method-only events (we just want attrs)
+    for attr in set(common_attrs):
+        # properties do not count as methods and will not be excluded
+        if ismethod(getattr(first_layer.__class__, attr, None)):
+            common.discard(attr)
+
+    return common
+
+
+def experimental_link_layers(
+    layers: Iterable['Layer'], attributes: Iterable[str] = ()
+):
+    """Link ``attributes`` between all layers in ``layers``.
+
+    This essentially performs the following operation:
+
+    .. code-block:: python
+
+       for lay1, lay2 in permutations(layers, 2):
+           for attr in attributes:
+              lay1.events.<attr>.connect(_set_lay2_<attr>)
+
+    Recursion is prevented by checking for value equality prior to setting.
+
+    Parameters
+    ----------
+    layers : Iterable[napari.layers.Layer]
+        The set of layers to link
+    attributes : Iterable[str], optional
+        The set of attributes to link.  If not provided (the default),
+        *all*, event-providing attributes that are common to all ``layers``
+        will be linked.
+
+    Raises
+    ------
+    ValueError
+        If any of the attributes provided are not valid "event-emitting"
+        attributes, or are not shared by all of the layers provided.
+
+    Examples
+    --------
+    >>> data = np.random.rand(3, 64, 64)
+    >>> viewer = napari.view_image(data, channel_axis=0)
+    >>> experimental_link_layers(viewer.layers)
+    """
+    from itertools import permutations, product
+
+    from ...utils.misc import get_equality_operator
+
+    valid_attrs = _get_common_evented_attributes(layers)
+
+    # now, ensure that the attributes requested are valid
+    attr_set = set(attributes)
+    if attributes:
+        extra = attr_set - valid_attrs
+        if extra:
+            raise ValueError(
+                "Cannot link attributes that are not shared by all "
+                f"layers: {extra}. Allowable attrs include:\n{valid_attrs}"
+            )
+    else:
+        # if no attributes are specified, ALL valid attributes are linked.
+        attr_set = valid_attrs
+
+    # now, connect requested attributes between all requested layers.
+    for (lay1, lay2), attr in product(permutations(layers, 2), attr_set):
+
+        key = (id(lay1), id(lay2), attr)
```

Comment:
Can you add a comment in the code here to the effect of
`# if the layers and attribute are already linked then ignore`